UnityProject/Assets/Scripts/Objects/Climbable.cs

```diff
@@ -0,0 +1,60 @@
+using UnityEngine;
+
+namespace Objects
+{
+	public class Climbable : MonoBehaviour, ICheckedInteractable<MouseDrop>
+	{
+		public float ClimbTime = 1.0f;
+
+		public bool WillInteract(MouseDrop interaction, NetworkSide side)
+		{
+			PlayerSync playerSync;
+			CustomNetTransform netTransform;
+			if (interaction.UsedObject.TryGetComponent(out playerSync))
+			{
+				if (playerSync.IsMoving || playerSync.playerMove.IsBuckled) return false;
+
+				// Do a sanity check to make sure someone isn't dropping the shadow from like 9000 tiles away.
+				float mag = (interaction.TargetObject.transform.position - playerSync.ServerPosition).magnitude;
```

Comment:
`playerSync.ServerPosition` will need to be client position when the side is the client

Reply:
wouldn't that allow for players to cheat? or is that variable stored on the server and not the client itself?

Reply:
I think I've solved what you wanted

Reply:
I've made a suggestion for it